website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,268 @@
+- What does the output mean?  
+- What is the [metric type](https://prometheus.io/docs/concepts/metric_types/#metric-types)? For example: `Gauge`/`Counter`/`Histogram` etc.
+- What unique labels does the metric use? if applicable. Please note (https://prometheus.io/docs/practices/naming/#labels).
+
+The `Help` message can be used to create auto-generated documentation, Like it's done here for example [KubeVirt metrics](https://github.com/kubevirt/kubevirt/blob/main/docs/metrics.md) and generated by [KubeVirt metrics doc generator](https://github.com/kubevirt/kubevirt/blob/main/tools/doc-generator/doc-generator.go).
+
+We recommend to auto-generated metrics documentation and save it in your operator repository, to a location like to `/docs/monitoring/`,  
+so that the users can find the information about your operator metrics easily. 
+
+#### Prometheus Recording Rules Naming
+As per [Prometheus](https://prometheus.io/docs/prometheus) documentation, [Recording rules](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/#recording-rules) allow you to pre-compute frequently needed or computationally expensive expressions and save their result as a new set of time series.
+
+**Note:** The Prometheus recording rules appear in Prometheus UI as metrics.
+In order to easily identify your operator recording rules, their name should follow the same naming guidelines as the metrics.
+
+### Prometheus Alerts Guidelines
+Clear and actionable alerts are a key component of a smooth operational
+experience and will result in a better experience for the end users.
+
+The following guidances aim to align alert naming, severities, labels, etc.
+in order to avoid alerts fatigue for administrators.
+
+#### Recommended Reading
+
+A list of references on good alerting practices:
+
+* [Google SRE Book - Monitoring Distributed Systems](https://sre.google/sre-book/monitoring-distributed-systems/)
+* [Prometheus Alerting Documentation](https://prometheus.io/docs/practices/alerting/)
+* [Alerting for Distributed Systems](https://www.usenix.org/sites/default/files/conference/protected-files/srecon16europe_slides_rabenstein.pdf)
+
+#### Alert Ownership
+
+Individual operator authors are responsible for writing and maintaining alerting rules for
+their components, i.e. their operators and operands.
+
+Operator authors should also take into consideration how their components interact with
+existing monitoring and alerting.
+
+As an example, if your operator deploys a service which creates one or more `PersistentVolume` resources,
+and these volumes are expected to be mostly full as part of normal operation, it's likely
+that this will cause unnecessary `KubePersistentVolumeFillingUp` alerts to fire.
+
+You should work to find a solution to avoid triggering these alerts if they are not actionable.
```

Comment:
This seems relative to [inhibition](https://prometheus.io/docs/alerting/latest/alertmanager/#inhibition) and [silences](https://prometheus.io/docs/alerting/latest/alertmanager/#silences).

I think it necessary to notify operator authors of the usage on these two technologies where they will greatly improve the efficiency of alerting.
Also, not sure if it can be good to provide [a group manifest template](https://prometheus.io/docs/alerting/latest/configuration/#example) to focus on alert management over operators.

Reply:
@Kavinjsir  I'll be happy if you can add this part to the doc after it is merged.
I would consider this as an advanced topic.